### PR TITLE
Packaging overhaul

### PR DESCRIPTION
**Pull request by zmc**

### CODE REVIEW

File: teuthology/task/tests/__init__.py [hunk 1]
```diff
@@ -44,8 +44,10 @@ def pytest_generate_tests(self, metafunc):
...
 
     # log the outcome of each test
-    def pytest_runtest_makereport(self, __multicall__, item, call):
-        report = __multicall__.execute()
+    @pytest.hookimpl(hookwrapper=True)
+    def pytest_runtest_makereport(self, item, call):
+        outcome = yield
+        report = outcome.get_result()
 
         # after the test has been called, get it's report and log it
```
Comment (dmick) [inline teuthology/task/tests/__init__.py:52]: can we fix that "it's" while we're here.  grinds my gears.

File: bootstrap [hunk 2]
```diff
@@ -157,7 +157,7 @@ fi
 find teuthology -name '*.pyc' -exec rm {} \;
 
 # Install teuthology in editable mode
-./$VENV/bin/pip install --editable .
+./$VENV/bin/pip install --editable '.[test]'
```
Comment (dmick) [inline bootstrap]: I've spent about five minutes searching the pip/PEP docs for wtf .[test] might mean (and still haven't found it).  Could you add a comment with at least the name of the concept for the next poor soul?

**Comment by kshtsk** [comment]
Looks interesting, I gonna try it.

File: bootstrap (hunk 2 above)
Comment (zmc) [reply bootstrap]: This is similar to doing `pip install teuthology[test]`, and indicates we want to install [optional dependencies](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies) as specified [in setup.cfg](https://github.com/ceph/teuthology/pull/1737/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52R99-R110). I can add a comment to that effect

File: teuthology/task/tests/__init__.py (hunk 1 above)
Comment (zmc) [reply teuthology/task/tests/__init__.py]: done

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by zmc** [comment]
@kshtsk the failure appears to just be an issue of missing packages

**Comment by kshtsk** [comment]
> @kshtsk the failure appears to just be an issue of missing packages

right

**Comment by kshtsk** [comment]
but because the run_job is changed, I think we still need to see the real run_job still works, and maybe we just need to schedule later to get the package?

**Comment by zmc** [comment]
> but because the run_job is changed, I think we still need to see the real run_job still works, and maybe we just need to schedule later to get the package?

We do execute `run_job` in [the new "integration" check](https://github.com/ceph/teuthology/runs/5978373989?check_suite_focus=true) - if you expand the "test using docker-compose" section, you can see that we schedule a job that the dispatcher picks up and runs. But of course I would not mind seeing @susebot run its tests successfully as well

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@zmc do you have an idea why it cannot find anything? https://ceph-ci.suse.de/job/teuthology-deploy-ovh/513/artifact/logs/teuthology-suite.log/*view*/

**Comment by kshtsk** [comment]
> @kshtsk It appears your tool is looking for octopus packages, and I don't see anything recent available: https://shaman.ceph.com/builds/ceph/octopus/

```2022-04-12 13:33:51,820.820 INFO:teuthology.suite.run:kernel sha1: distro
2022-04-12 13:33:54,123.123 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph octopus -> b51d866559c7cde15d3b75ee17bbba8fdf2584be
2022-04-12 13:33:54,124.124 INFO:teuthology.suite.run:ceph sha1: b51d866559c7cde15d3b75ee17bbba8fdf2584be
2022-04-12 13:33:54,124.124 INFO:teuthology.suite.run:skipping ceph package verification
2022-04-12 13:33:54,602.602 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph.git octopus -> b51d866559c7cde15d3b75ee17bbba8fdf2584be
2022-04-12 13:33:55,267.267 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph.git octopus -> b51d866559c7cde15d3b75ee17bbba8fdf2584be
2022-04-12 13:33:55,267.267 INFO:teuthology.suite.run:ceph branch: octopus b51d866559c7cde15d3b75ee17bbba8fdf2584be
```

and from here https://shaman.ceph.com/builds/ceph/octopus/b51d866559c7cde15d3b75ee17bbba8fdf2584be/
I see https://shaman.ceph.com/builds/ceph/octopus/b51d866559c7cde15d3b75ee17bbba8fdf2584be/default/299324/

**Comment by zmc** [comment]
> > @kshtsk It appears your tool is looking for octopus packages, and I don't see anything recent available: https://shaman.ceph.com/builds/ceph/octopus/
> 
> ```
> 2022-04-12 13:33:54,123.123 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph octopus -> b51d866559c7cde15d3b75ee17bbba8fdf2584be
> 2022-04-12 13:33:54,124.124 INFO:teuthology.suite.run:ceph sha1: b51d866559c7cde15d3b75ee17bbba8fdf2584be
> 2022-04-12 13:33:54,124.124 INFO:teuthology.suite.run:skipping ceph package verification
> 2022-04-12 13:33:54,602.602 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph.git octopus -> b51d866559c7cde15d3b75ee17bbba8fdf2584be
> 2022-04-12 13:33:55,267.267 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph.git octopus -> b51d866559c7cde15d3b75ee17bbba8fdf2584be
> 2022-04-12 13:33:55,267.267 INFO:teuthology.suite.run:ceph branch: octopus b51d866559c7cde15d3b75ee17bbba8fdf2584be
> ```
> 
> and from here https://shaman.ceph.com/builds/ceph/octopus/b51d866559c7cde15d3b75ee17bbba8fdf2584be/ I see https://shaman.ceph.com/builds/ceph/octopus/b51d866559c7cde15d3b75ee17bbba8fdf2584be/default/299324/

Right, so if you click on the "View Repos ..." button up top, you'll find an empty list. Contrast this with a more recent build I found at random: https://shaman.ceph.com/builds/ceph/master/d104ae541c9785cf2d10bc42249ceb4367f94dbe/default/301232/

I believe we keep builds for 14d.

**Comment by kshtsk** [comment]
and this is the latest sha1, so does it mean no new builds for octopus so far... hm... wondering which branch/suite can be used for that

**Comment by zmc** [comment]
> and this is the latest sha1, so does it mean no new builds for octopus so far... hm... wondering which branch/suite can be used for that

Why not master?

**Comment by kshtsk** [comment]
> > and this is the latest sha1, so does it mean no new builds for octopus so far... hm... wondering which branch/suite can be used for that
> 
> Why not master?

it is probably makes sense, if it is more stable

File: teuthology/dispatcher/supervisor.py [hunk 3]
```diff
@@ -31,7 +31,6 @@ def main(args):
 
     verbose = args["--verbose"]
     archive_dir = args["--archive-dir"]
-    teuth_bin_path = args["--bin-path"]
```
Comment (kamoltat) [inline teuthology/dispatcher/supervisor.py:34]: 👍 for refractoring

**Review by kamoltat — APPROVED** [review]
LGTM

**Pull request merged**